libraries/ms-helper/src/x86/adc_periodic_reader.c

```diff
@@ -0,0 +1,82 @@
+#include "adc_periodic_reader.h"
+#include "adc.h"
+#include "delay.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "gpio_mcu.h"
+#include "interrupt.h"
+#include "log.h"
+#include "misc.h"
+#include "soft_timer.h"
+#include "status.h"
+
+#define TIMER_INTERVAL 50
+
+static AdcPeriodicReaderStorage s_storage[NUM_PERIODIC_READER_IDS];
+
+void prv_callback(SoftTimerId timer_id, void *context) {
+  for (size_t i = 0; i < NUM_PERIODIC_READER_IDS; i++) {
+    uint16_t data = 0;
+    AdcChannel channel;
+    adc_get_channel(s_storage[i].address, &channel);
+    adc_read_converted(channel, &data);
+    s_storage[i].data = data;
+    if (s_storage[i].activated) {
+      s_storage[i].callback(s_storage[i].data, i, s_storage[i].context);
+    }

```

Comment:
Should probably not have read the data if this reader was not activated

Reply:
This will not read data if the reader is not activated?